packages/routers/src/types.tsx

```diff
@@ -125,6 +125,13 @@ export type DefaultRouterOptions<RouteName extends string = string> = {
    * If not specified, usually the first route is used.
    */
   initialRouteName?: RouteName;
+  /**
+   A function returning a state, which may be set after modifying the routes name.
+   */
+  getStateForRouteNamesChange?: (
+    state: NavigationState
+  ) => PartialState<NavigationState> | undefined;
+  linking?: any;
```

Comment:
These don't look right 🤔

`getStateForRouteNamesChange` is an option for `useNavigationBuilder` and `linking` is not for router.